src/PagerView.tsx

```diff
@@ -233,3 +233,9 @@ export const PagerView = React.forwardRef<PagerViewInternal, PagerViewProps>(
     return <PagerViewInternal {...rest} useLegacy={!useNext} ref={ref} />;
   }
 );
+
+// React.forwardRef does not type returned component properly, thus breaking Ref<MyComponent> typing.
+// One way to overcome this is using separate typing for component "interface",
+// but that breaks backward compatibility in this case.
+// Approach of merging type is hacky, but produces a good typing for both ref attributes and component itself.
+export type PagerView = PagerViewInternal & typeof PagerView;
```

Comment:
I'm addressing 2 problems here:
1. PagerView is a value, not a type, since we're using forwardRef. Therefore, it cannot be used as type at all, e.g. in `useRef<PagerView>`, current error is that PagerView is a value that's attempted to be used as a type. This could be mitigated by simply: `export type PagerView = typeof PagerView`. We can create types with the same names as values, TS chooses type/value basing on usage context (similar as built-in class-type mechanism in TS)
2. forwardRef does not type component "interface" at all. `typeof PagerView` does not show any ref based methods.

To overcome this, there are two approaches:
1. Export separate typing for component interface, e.g. `PagerViewInterface/Ref` and make all users switch to using it in useRef
2. The one I'm using above, that has a drawback of mixing two types, but preserves backward compatibility.

I suggest the second one, because lib user facing interface shouldn't be affected whether class or function component is used internally.

Reply:
@troZee